libs/dyn/pattern.go

```diff
@@ -69,7 +69,7 @@ func (c anyKeyComponent) visit(v Value, prefix Path, suffix Pattern, opts visitO
 			return InvalidValue, err
 		}
 
-		m.Set(pk, nv) //nolint:errcheck
```

Comment:
Is it an equivalent change? Won't pk now lose the location information if it had any?